avogadro/io/CMakeLists.txt

```diff
@@ -51,7 +52,7 @@ endif()
 
 avogadro_add_library(AvogadroIO ${HEADERS} ${SOURCES})
 
-target_link_libraries(AvogadroIO LINK_PUBLIC AvogadroCore LINK_PRIVATE jsoncpp)
+target_link_libraries(AvogadroIO LINK_PUBLIC AvogadroCore LINK_PRIVATE jsoncpp LINK_PRIVATE struct)
```

Comment:
You don't need to repeat LINK_PRIVATE, everything after will use private linking.